# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/security/DOMSecurityManager.cpp
# Commit: 6dffdf63221a
# Full Hash: 6dffdf63221aa03f7c52e3aaf408b485ddcb6bbc
# Author: Christoph Kerschbaumer <ckerschb@christophkerschbaumer.com>
# Date: 2019-10-23 21:54:06
# Description:
#   Bug 1590777: Add Null check for referrerinfo within ParseCSPAndEnforceFrameAncestorCheck. r=tnguyen
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D50239
# ==============================================================================

diff -r 2abea3ccd7b8 -r 6dffdf63221a dom/security/DOMSecurityManager.cpp
--- a/dom/security/DOMSecurityManager.cpp	Wed Oct 23 15:17:21 2019 +0000
+++ b/dom/security/DOMSecurityManager.cpp	Wed Oct 23 15:21:48 2019 +0000
@@ -143,7 +143,9 @@
 
   nsCOMPtr<nsIReferrerInfo> referrerInfo = httpChannel->GetReferrerInfo();
   nsAutoString referrerSpec;
-  referrerInfo->GetComputedReferrerSpec(referrerSpec);
+  if (referrerInfo) {
+    referrerInfo->GetComputedReferrerSpec(referrerSpec);
+  }
   uint64_t innerWindowID = loadInfo->GetInnerWindowID();
 
   rv = csp->SetRequestContextWithPrincipal(resultPrincipal, selfURI,
